sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -727,13 +729,14 @@ class HashFullJoinIterator(
         }
       }
     }
-    builtSideTracker = withResource(updatedTrackingTable) { newTab =>
-      withResource(GpuColumnVector.from(newTab, Array[DataType](DataTypes.BooleanType))) { cb =>
-        val lazyBatch = LazySpillableColumnarBatch(cb, spillCallback, "tracking_batch")
-        lazyBatch.allowSpilling()
-        Some(lazyBatch)
-      }
+    val newScb = withResource(updatedTrackingTable) { _ =>
+      SpillableColumnarBatch(
+        GpuColumnVector.from(updatedTrackingTable, Array[DataType](DataTypes.BooleanType)),
+        SpillPriorities.ACTIVE_ON_DECK_PRIORITY, spillCallback)
     }
+    // Close old spillable batch and replace with updated one.
+    builtSideTracker.foreach(_.close())
```

Comment:
newScb is leaked if this throws.  Ideally `builtSideTracker` should be closed during the computation of builtTrackingTable.

Reply:
Agreed.  I realized the same thing and I'm already testing the commit to fix it.